src/components/Panel/Panel.tsx

```diff
@@ -0,0 +1,101 @@
+/** @jsx h */
+
+import { h } from 'preact';
+import { useState, useEffect, useRef } from 'preact/hooks';
+import cx from 'classnames';
+import Template from '../Template/Template';
+import { PanelCSSClasses, PanelTemplates } from '../../widgets/panel/panel';
+import { RenderOptions } from '../../types';
+
+type PanelComponentCSSClasses = Omit<
+  {
+    [TKey in keyof PanelCSSClasses]: string;
+  },
+  // `collapseIcon` is only used in the default templates of the widget
+  'collapseIcon'
+>;
+
+interface PanelProps {
+  hidden: boolean;
+  collapsible: boolean;
+  isCollapsed: boolean;
+  data: RenderOptions;
+  cssClasses: PanelComponentCSSClasses;
+  templates: PanelTemplates;
+  bodyElement: HTMLElement;
+}
+
+function Panel(props: PanelProps) {
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(props.isCollapsed);
+  const [isControlled, setIsControlled] = useState<boolean>(false);
+  const bodyRef = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
+    if (!bodyRef.current) {
+      return;
+    }
+
+    bodyRef.current.appendChild(props.bodyElement);
```

Comment:
I'm not sure if simply appending is correct here, do you not need to clear the DOM inside the body before adding a new bodyElement? Seems like we didn't handle that before either though

Reply:
dealt with before I could submit my review 🐭 👒